src/v/cluster/commands.h

```diff
@@ -264,6 +270,34 @@ using update_user_cmd = controller_command<
   model::record_batch_type::user_management_cmd,
   serde_opts::adl_and_serde>;
 
+using create_role_cmd = controller_command<
+  security::role_name,
+  security::role,
```

Comment:
I think it is generally a good idea to completely ignore either the key or the value and create a _versioned_ structure for the payload of each command to future proof yourself against needing to add additional context to the command. the `rename` command is a good example. sure, using key=from-name and value=to-name works now, but what if you need some extra state later? you'd probably have to add a replacement command.

Reply:
@mmaslankaprv is there a new/better way to handle this?

Reply:
TBH we don't use the key for anything AFAIK, so I would rather just have a value and that would mean people would make their own structs for each commands, so you'd get the serde benefits. There are also things like this serde_only tag which should be the default.

I also don't love this special batch type pattern, I would rather just have the key or a header specify who owns a controller command

Reply:
`serde_only` is the default...noticed that yesterday and meant to remove the param in this PR.

>create a versioned structure for the payload

that makes good sense, thanks. I see some examples of that pattern in nearby code now.

Reply:
@dotnwat - this is done if you're interested in taking another look